Classes/Private/Libxml2/Data/CommentNode.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+extension Libxml2 {
+    /// Comment nodes use to hold HTML comments like this: <!-- This is a comment -->
+    ///
+    class CommentNode: Node {
+
+        var text: String
+
+        init(text: String) {
+            self.text = text
+
+            super.init(name: "text")
+        }
+
+        override func customMirror() -> Mirror {
+            return Mirror(self, children: ["type": "text", "name": name, "text": text, "parent": parent.debugDescription], ancestorRepresentation: .Suppressed)
+        }
+
+        /// Node length.
+        ///
+        override func length() -> Int {
+            return text.characters.count
+        }
+
+        override func deleteCharacters(inRange range: NSRange) {
+
+            guard let textRange = text.rangeFromNSRange(range) else {
+                fatalError("The specified range is out of bounds.")
+            }
+
+            text.removeRange(textRange)
+        }
+
+        override func replaceCharacters(inRange range: NSRange, withString string: String) {
+
+            guard let textRange = text.rangeFromNSRange(range) else {
+                fatalError("The specified range is out of bounds.")
+            }
+
+            text.replaceRange(textRange, with: string)
+        }
+
+        override func split(forRange range: NSRange) {
```

Comment:
This method shouldn't be present in comment nodes.  I can see this will complicate my initial design a bit, so how to remove this cleanly is open for discussion.